File: ClassicalDynamics/NewtonsLowsOfMotion/UniformMotion/TypeScript/Unim2D.ts
class Unim1D {
    private locationX: number = 0;
    private velocityX: number = 0;
    protected deltaTime: number = 0;
    private totalTime: number = 0;
    protected steps: number = 0;
    /**
     * @constructor
     * @param initX Initial value of x coordinate
     * @param initVX Initial value of velocity x
     * @param deltaTime incremental time
     * @param totalTime total time
     */
    public constructor(initX: number, initVX: number, deltaTime: number, totalTime: number) {
        this.locationX = initX;
        this.velocityX = initVX;
        this.deltaTime = deltaTime;
        this.totalTime = totalTime;
        this.steps = Math.floor(this.totalTime / this.deltaTime);
    }

    /**
     * 
     * @param nowTime now time
     */
    protected calcX(nowTime: number): number {
        return this.locationX + this.velocityX * nowTime;
    }

    public dataWrite(): any {
        let nowlocX : number = 0, nowTime : number = 0;
        for(let count = 1; count <= this.steps; count++){
            nowlocX = this.calcX(nowTime);
            console.log(nowTime, nowlocX);
            nowTime = nowTime + this.deltaTime;
        }
    }
}

class Unim2D extends Unim1D {
    private locationY : number = 0;
    private velocityY : number = 0;
    /**
     * @constructor
     * @param initX Initial value of x coordinate
     * @param initY Initial value of y coordinate
     * @param initVX Initial value of velocity x
     *  @param initVY Initial value of velocity y
     * @param deltaTime incremental time
     * @param totalTime total time
     */
    public constructor(initX: number, initY : number, initVX: number, initVY : number,deltaTime: number, totalTime: number) {
        super(initX, initVX, deltaTime, totalTime);
        this.locationY = initY;
        this.velocityY = initVY;
    }

    /**
     * 
     * @param nowTime now time
     */
    private calcY(nowTime : number) : number {
        return this.locationY + this.velocityY * nowTime;
    }

    public dataWrite() : void {
        let nowlocX = 0, nowlocY = 0;
        let nowTime = 0;
        for(let count = 0; count < this.steps; count++){
            nowlocX = this.calcX(nowTime);
            nowlocY = this.calcY(nowTime);
            console.log(nowlocX, nowlocY);
            nowTime = nowTime + this.deltaTime;
        } 
    }
}

let render = () => {
    let initX : number = 0.0 //座標xの初期値
    let initY = 0.0//座標xの初期値
    let initVX : number= 2.0 //速度xの初期値
    let initVY = 5.0//速度yの初期値
    let deltaTime : number = 0.1 //経過時間の値
    let totalTime : number = 4.0 //総合時間
    
    let unim2d = new Unim2D(initX, initY,initVX, initVY,deltaTime, totalTime);
    unim2d.dataWrite();
}

render();

File: ClassicalDynamics/NewtonsLowsOfMotion/UniformMotion/Python/Unim2D.py
import math
import matplotlib.pyplot as plt
class Unim1D:
    #変数の初期化(constructor)
    def __init__(self, initX, initVX, deltaTime, totalTime):
        self.locationX = initX
        self.velocityX = initVX
        self.deltaTime = deltaTime
        self.totalTime = totalTime
        self.steps = int(self.totalTime / self.deltaTime)
    
    def calcX(self, nowTime : float) -> float:
        return self.locationX + self.velocityX * nowTime   

    def dataWrite(self):
        nowlocX, nowTime = 0, 0
        dataX, dataY = [], []
        for _count in range(self.steps):
            nowlocX = self.calcX(nowTime)
            #print(nowTime, nowlocX)
            dataX.append(nowTime)
            dataY.append(nowlocX)
            nowTime = nowTime + self.deltaTime
        
        plt.plot(dataX, dataY, 'go')
        plt.savefig("unim1D.png")

class Unim2D(Unim1D):
    def __init__(self, initX, initY,initVX, initVY,deltaTime, totalTime):
        Unim1D.__init__(self, initX, initVX, deltaTime, totalTime)
        self.locationY = initY
        self.velocityY = initVY
    
    def calcY(self, nowTime : float) -> float:
        return self.locationY + self.velocityY * nowTime

    def dataWrite(self):
        nowlocX, nowlocY = 0, 0
        nowTime = 0
        dataX, dataY = [], []
        for _count in range(self.steps):
            nowlocX = self.calcX(nowTime)
            nowlocY = self.calcY(nowTime)
            #print(nowlocX, nowlocY)
            dataX.append(nowlocX)
            dataY.append(nowlocY)
            nowTime = nowTime + self.deltaTime
        
        plt.plot(dataX, dataY, 'go')
        plt.savefig("unim2D.png")

if __name__ == "__main__":
    initX = 0.0 #座標xの初期値
    initY = 0.0#座標xの初期値
    initVX = 2.0 #速度xの初期値
    initVY = 5.0#速度yの初期値
    deltaTime = 0.1 #経過時間の値
    totalTime = 4.0 #総合時間

    unim2d = Unim2D(initX, initY,initVX, initVY,deltaTime, totalTime)
    unim2d.dataWrite()

